infra/docker/web/Dockerfile

```diff
@@ -0,0 +1,101 @@
+FROM php:7.4-fpm-alpine
+
+EXPOSE 80
+
+# Create some environment variables if not present.
+# TBD - We're going to be injecting a lot of values via GitHub Actions.
+
+# Configure timezone
+RUN ln -s /usr/share/zoneinfo/Etc/UTC /etc/localtime
+
+# Install Libraries
+RUN apk add --update --no-cache \
+    libgd \
+    imagemagick \
+    html2text \
+    nginx \
+    freetype \
+    postgresql \
+    tidyhtml
+
+# Install PHP Extensions not already present in alpine release
+# RUN docker-php-ext-install -j$(nproc) \
+#     opcache \
+#     pgsql \
+#     tidy \
+#     gd
+
+# Memcached PHP lib
+RUN set -xe && \
+    cd /tmp/ && \
+    apk add --no-cache --update --virtual .phpize-deps $PHPIZE_DEPS && \
+    apk add --no-cache --update --virtual .memcached-deps zlib-dev libmemcached-dev cyrus-sasl-dev libpng-dev \
+    libjpeg-turbo-dev \
+    freetype-dev \ 
+    postgresql-dev \
+    libmemcached-libs \
+    zlib \
+    tidyhtml-dev && \
+# Install igbinary (memcached's deps)
+    pecl install igbinary && \
+# Install memcached
+    ( \
+        pecl install --nobuild memcached && \
+        cd "$(pecl config-get temp_dir)/memcached" && \
+        phpize && \
+        ./configure --enable-memcached-igbinary && \
+        make -j$(nproc) && \
+        make install && \
+        cd /tmp/ \
+    ) && \
+# Enable PHP extensions
+    docker-php-ext-enable igbinary memcached && \
+    rm -rf /tmp/* && \
+    docker-php-ext-install -j$(nproc) \
+    opcache \
+    pgsql \
+    tidy \
+    gd && \
+    apk del .memcached-deps .phpize-deps
```

Comment:
I personally advocate helper scripts to install complicated stuff like this (see `install-xdiff.sh`), but this is obviously not blocking.